cmd/plugin/delete_old_txt.go

```diff
@@ -0,0 +1,203 @@
+		return nil
+	}
+
+	providerFactory, err := provider.NewFactory(k8sClient, dynClient, provider.RegisteredDefaultProviders(), endpoint.NewAuthoritativeDNSRecordProvider)
+	if err != nil {
+		fmt.Printf("failed to create provider factory: %v\n", err)
+		return nil
+	}
+
+	// empty config to list all records
+	endpointProvider, err := providerFactory.ProviderForSecret(ctx, secret, provider.Config{})
+	if err != nil {
+		fmt.Printf("failed to get provider: %v\n", err)
+		return nil
+	}
+
+	endpoints, err := endpointProvider.Records(ctx)
+	if err != nil {
+		fmt.Printf("failed to get endpoints: %v\n", err)
+		return nil
+	}
+
+	// only old TXTs created by us should be left
+	endpoints = slice.Filter(endpoints, func(e *externaldns.Endpoint) bool {
+		if e != nil &&
+			e.RecordType == externaldns.RecordTypeTXT &&
+			strings.HasSuffix(e.DNSName, deleteOldTXTCMDFlags.domain) {
+
+			// we will always have at least one target
+			if len(e.Targets) == 0 {
+				return false
+			}
+
+			// two layers of checks to make sure we don't delete anything needed
+
+			// owner and version in targets
+			owner, epVersion, epLabels, err := registry.NewLabelsFromString(e.Targets[0], []byte{})
+			if err != nil {
+				fmt.Printf("failed to extract owner labels: %v\n", err)
+				return false
+			}
+
+			// happens when there was only one owner
+			if owner == "" {
+				owner = epLabels[externaldns.OwnerLabelKey]
+			}
+
+			// it is old only if it has owner and doesn't have version
+			if owner != "" && epVersion == "" {
+				// this should be redundant,
+				// but it might happen that they have TXTs from external dns that are not from our operator
+				// make sure the name of endpoint matches our old pattern
+				mapper := registry.NewExternalDNSAffixNameMapper("kuadrant-", "", "wildcard")
+				epName, recordType := mapper.ToEndpointName(e.DNSName, epVersion)
+
+				// if they aren't empty - it was our record
+				return epName != "" && recordType != ""
+			}
+			return false
+		}
+		return false
+	})
+
+	fmt.Printf("filtered endpoints len %d\n", len(endpoints))
+	for _, ep := range endpoints {
+		fmt.Printf("ep: %s\n", ep.DNSName)
+	}
+
+	// display what is about to be deleted
+	fmt.Printf("TXT records (%d) to be deleted:\n", len(endpoints))
+	for _, txtRecord := range endpoints {
+		fmt.Printf("%s\t IN\t %s\t %s\n", txtRecord.DNSName, txtRecord.RecordType, txtRecord.Targets)
+	}
+	fmt.Printf("Do you want to proceed? [Y/N]\n")
+	reader := bufio.NewReader(os.Stdin)
+	answer, _ := reader.ReadString('\n')
+	answer = strings.TrimSpace(strings.ToLower(answer))
+
+	switch answer {
+	case "y":
+		//delete
+		fmt.Printf("deleting old TXT records...\n")
+		err = endpointProvider.ApplyChanges(ctx, &plan.Changes{
+			Delete: endpoints,
+		})
+		if err != nil {
+			fmt.Printf("failed to delete old TXT records: %v\n", err)
+		}
+		return nil
+	case "n":
+		// do nothing
+		return nil
+	default:
+		fmt.Printf("unknown answer %s\n", answer)
+		return nil
+	}
```

Comment:
It is common for a cli to that asks a "can we proceed" question to default to "no". In this case they would get an unknown answer response, over we are cancelling. 

For the no case it might be good to have some feedback to the user saying the process is being cancelled.

Reply:
fair. Also added a message for a "yes" to make them know we are done